hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/ControllerManagement.java

```diff
@@ -114,6 +115,17 @@
     List<Action> findActionByTargetAndActive(@NotNull Target target);
 
     /**
+     * Retrieves oldest {@link Action} that is active and assigned to a
+     * {@link Target}.
+     *
+     * @param target
+     *            the target to retrieve the actions from
+     * @return a list of actions assigned to given target which are active
+     */
+    @PreAuthorize(SpringEvalExpressions.IS_CONTROLLER)
+    Optional<Action> findOldestActionByTargetAndActive(@NotNull Target target);
```

Comment:
A better name would be `findOldestActiveActionByTarget`.  You don't have to follow the Spring-Data naming conventions here. Take also `findActionByTargetAndActive` into account => `findActiveActionByTarget`